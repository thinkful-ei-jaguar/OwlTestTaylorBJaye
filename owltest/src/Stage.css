* {
    box-sizing: border-box;
}

.nodisplay {
    display: none;
}

.graybackground {
    display: inline-flex;
    flex-direction: row;
    margin: 100px;
    margin-right: auto;
    margin-left: auto;
}

.person {
    display: flex;
    flex-direction: column;
}

p {
    color: black;
    font-size: 25px;
}

img {
    width: 30px;
    height: 30px;
    border-radius: 100%;
    font-size: 50px;
}

