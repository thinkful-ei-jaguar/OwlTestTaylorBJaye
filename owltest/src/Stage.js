import React from 'react';
import './Stage.css';

export default function Stage (props) {

    if(props.onStage) {
        return (
            <div className = 'graybackground'>
                <div className = 'person'>
                    <p>{props.name}</p>
                    <img src={props.avatar} alt = {props.name}></img>
                </div>
            </div>
        )
    }
    else{
        return (
            <div className = 'nodisplay'></div>
        )
    }
}