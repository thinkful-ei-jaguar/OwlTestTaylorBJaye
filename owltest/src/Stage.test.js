import React from 'react';
import ReactDOM from 'react-dom';
import Stage from './Stage';
import renderer from 'react-test-renderer';

describe('Stage Component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Stage 
            id= {123}
            name='Kitten'
            avatar= 'https://placekitten.com/200/287'
            inSession= {true}
            onStage= {true}
            />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the Stage UI as expected', () => {
        const foliage = renderer
        .create(<Stage 
            id= {123}
            name='Kitten'
            avatar= 'https://placekitten.com/200/287'
            inSession= {true}
            onStage= {true}
            />)
        .toJSON();
        expect(foliage).toMatchSnapshot();
    });
});