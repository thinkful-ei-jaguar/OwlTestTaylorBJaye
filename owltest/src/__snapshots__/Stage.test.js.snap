// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage Component renders the Stage UI as expected 1`] = `
<div
  className="graybackground"
>
  <div
    className="person"
  >
    <p>
      Kitten
    </p>
    <img
      alt="Kitten"
      src="https://placekitten.com/200/287"
    />
  </div>
</div>
`;
